src/finders/ApiGatewayAuthorizerIdFinder.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * This finder try to find the Authorizer Id for a given ApiGatewayName/AuthorizerName
+ */
+class ApiGatewayAuthorizerIdFinder {
+
+  async find(name) {
+    const serverless = this.serverless
+    const provider = this.provider
+    let apiGatewayName
+    let apiGatewayAuthorizerName
+
+    if (name) {
+      apiGatewayName = name.split("/")[0]
+      apiGatewayAuthorizerName = name.split("/")[1]
+    }
+
+    const response = await this.provider.request(
+      "APIGateway",
+      "getRestApis",
+      {
+        limit: "500"
+      }
+    )
+    if (response) {
+      this.apiGatewayIds = {}
+      for (const gateway of response.items) {
+        this.apiGatewayIds[gateway.name] = gateway.id
+      }
+    }
+
+    if (this.apiGatewayIds) {
+      let apiGatewayId
+      let apiGatewayAuthorizerId
+      if (apiGatewayName) {
+        apiGatewayId = this.apiGatewayIds[apiGatewayName]
```

Comment:
I think this would throw an error if no `name` was provided and the user's AWS account has one or more API Gateways, because in that case `apiGatewayName` would be `undefined`.

Reply:
If no Api Gateway, it go to the else branch and throw errors. 
If no name provide, it pick up the first API gateway.